torch/csrc/autograd/record_function.cpp

```diff
@@ -18,85 +18,178 @@ float sample_zero_one() {
   return dist(*gen);
 }
 
+class RecordFunctionCallback {
```

Comment:
nit: Would be nice to add some comments about this class, its public methods and some of the private member fields.

Reply:
added in https://github.com/pytorch/pytorch/pull/37491